policy/src/main/java/org/apache/brooklyn/policy/ha/ElectPrimaryPolicy.java

```diff
@@ -0,0 +1,223 @@
+        "propagate.primary.sensors");
+    
+    @Override
+    public void setEntity(@SuppressWarnings("deprecation") org.apache.brooklyn.api.entity.EntityLocal entity) {
+        super.setEntity(entity);
+        
+        checkAndMaybeAddEffector(entity);
+        checkQuorums(entity);
+        addSubscriptions(entity);
+        
+        Collection<? extends Object> sensorsToPropagate = config().get(PROPAGATE_PRIMARY_SENSORS);
+        if (sensorsToPropagate!=null) {
+            List<Sensor<?>> realSensors = MutableList.of();
+            for (Object s: sensorsToPropagate) {
+                if (s instanceof String) s = Sensors.newSensor(Object.class, (String) s);
+                if (s instanceof Sensor) {
+                    realSensors.add((Sensor<?>)s);
+                } else {
+                    throw new IllegalArgumentException("Config "+PROPAGATE_PRIMARY_SENSORS.getName()+" had invalid entry "
+                        + "'"+s+"'; expected string or sensor");
+                }
+            }
+            entity.enrichers().add(EnricherSpec.create(PropagatePrimaryEnricher.class)
+                .configure(PropagatePrimaryEnricher.PROPAGATE_EFFECTORS, false)
+                .configure(PropagatePrimaryEnricher.PROPAGATING, realSensors));
+        }
+    }
+
+    protected void checkAndMaybeAddEffector(@SuppressWarnings("deprecation") org.apache.brooklyn.api.entity.EntityLocal entity) {
+        String effName = config().get(EFFECTOR_NAME);
+        if (((EntityInternal)entity).getEffector(effName)==null) {
+            // effector not defined
+            if (config().getRaw(EFFECTOR_NAME).isAbsent()) {
+                log.debug("No effector '"+effName+"' present at "+entity+"; creating default");
+                // if not set, we can create the default
+                new ElectPrimaryEffector(config().getBag()).apply(entity);
+                
+            } else {
+                // otherwise it's an error, fail
+                throw new IllegalStateException("No such effector '"+effName+"' on "+entity);
+            }
+        }
+    }
+
+    private void checkQuorums(Entity entity) {
+        // set all quorums to 1 if not explicitly set
+        if ( ((ConfigurationSupportInternal)entity.config()).getRaw(StartableApplication.UP_QUORUM_CHECK).isAbsent() ) {
+            entity.config().set(StartableApplication.UP_QUORUM_CHECK, QuorumChecks.newInstance(1, 0.0, false));
+        }
+        if ( ((ConfigurationSupportInternal)entity.config()).getRaw(StartableApplication.RUNNING_QUORUM_CHECK).isAbsent() ) {
+            entity.config().set(StartableApplication.RUNNING_QUORUM_CHECK, QuorumChecks.newInstance(1, 0.0, false));
+        }
+    }
+
+    protected void addSubscriptions(Entity entity) {
+        String weightSensorName = config().get(PRIMARY_WEIGHT_NAME);
+        highlightTriggers("Listening for "+weightSensorName+" and service up, state on all children " + (entity instanceof Group ? "and members " : ""));
```

Comment:
it would only consider as candidates children OR members depending on `TARGET_MODE` (with default `AUTO` doing an `instanceof Group` check).  but it would subscribe to all.  have changed so it only subscribes to children OR members, and the reflects that.